src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -578,7 +578,7 @@ def _search_role_assignments(cli_ctx, assignments_client, definitions_client,
     # "atScope()" and "principalId eq '{value}'" query cannot be used together (API limitation).
     # always use "scope" if provided, so we can get assignments beyond subscription e.g. management groups
     if scope:
-        f = 'atScope()'
+        f = 'atScope()'  # atScope() excludes role assignments at subscopes
```

Comment:
More discussion on this enforced `atScope()`: https://github.com/Azure/azure-cli/issues/14302